cmd/iam-store.go

```diff
@@ -965,7 +958,13 @@ func (store *IAMStoreSys) PolicyDBUpdate(ctx context.Context, name string, isGro
 	var mp MappedPolicy
 	if !isGroup {
 		if userType == stsUser {
-			mp = cache.iamSTSPolicyMap[name]
+			var ok bool
+			mp, ok = cache.iamSTSPolicyMap[name]
+			if !ok {
+				// Attempt to load parent user mapping for STS accounts
+				store.loadMappedPolicy(context.TODO(), name, stsUser, false, cache.iamSTSPolicyMap)
+				mp = cache.iamSTSPolicyMap[name]
+			}
```

Comment:
2. There may still be an issue here. The bug being fixed here is that the stsPolicyMap does not contain the STS user's policy. This is fixed in the notification handler to load the mapping properly. However, there is a problem when the notification is missed by a node and the policy mapping is being updated (i.e. already exists and is being changed). In that case, a stale mapping would be present, and the policy DB update would not be correct.

Reply:
Isn't that expected to be fixed via refresh() routine? @donatello